### PR TITLE
Uaw 142 integration tests session

### PR DESCRIPTION
**Pull request by popsps**
Session controller: 100% code/branch coverage Session controller: 100% code/branch coverage

### CODE REVIEW

**Review by Samnj583 — DISMISSED** [review]

File: backend/authentication-microservice/src/main/java/com/ss/utopia/auth/controller/SessionController.java [hunk 1]
```diff
@@ -90,10 +94,12 @@ public User getSession(@AuthenticationPrincipal UserDetails currentUser) {
       return userService.getUserByUsername(currentUser.getUsername());
   }
 
+  // TODO: 2/6/2021 POST /api/sessions/users/user/logout
   @DeleteMapping
   @ResponseStatus(HttpStatus.NO_CONTENT)
   public void logout(HttpServletResponse response) {
-    final Cookie sessionCookie = userService.removeCookie();
+    final Cookie sessionCookie = userService.removeCookie()
```
Comment (WesleyClements) [inline backend/authentication-microservice/src/main/java/com/ss/utopia/auth/controller/SessionController.java:101]: Why are we throwing and error? Doesn't removeCookie just return a new Cookie object with a hard coded value and path? When would that ever throw an error?
See [constructor](https://docs.oracle.com/javaee/6/api/javax/servlet/http/Cookie.html#Cookie(java.lang.String,%20java.lang.String)) and [setPath](https://docs.oracle.com/javaee/6/api/javax/servlet/http/Cookie.html#setPath(java.lang.String)) documentation.

File: backend/authentication-microservice/src/main/java/com/ss/utopia/auth/service/UserService.java [hunk 2]
```diff
@@ -170,10 +169,15 @@ public User verifyOwnershipAndReturnOwner(User user, UserDetails currentUser) {
    *
    * @return
    */
-  public Cookie removeCookie() {
-    final Cookie sessionCookie = new Cookie("session", null);
-    sessionCookie.setPath("/");
-    return sessionCookie;
+  public Optional<Cookie> removeCookie() {
+    try {
+      final Cookie sessionCookie = new Cookie("session", null);
```
Comment (WesleyClements) [inline backend/authentication-microservice/src/main/java/com/ss/utopia/auth/service/UserService.java:151]: See my other comment.

File: backend/authentication-microservice/src/test/java/com/ss/utopia/auth/controller/SessionControllerIntegrationTest.java [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+  void tearDown() {
+  }
+
+  /**
+   * login an admin user to the auth service
+   *
+   * @return
+   * @throws Exception
+   */
+  private String login() throws Exception {
```
Comment (WesleyClements) [inline backend/authentication-microservice/src/test/java/com/ss/utopia/auth/controller/SessionControllerIntegrationTest.java:58]: Is this meant to be a test?

File: backend/authentication-microservice/src/main/java/com/ss/utopia/auth/entity/User.java [hunk 4]
```diff
@@ -131,4 +134,22 @@ public UserRole getRole() {
...
   }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) return true;
+    if (!(o instanceof User)) return false;
+    User user = (User) o;
+    PasswordEncoder passwordEncoder = new BCryptPasswordEncoder(11);
+
+    return getId().equals(user.getId()) && getUsername().equals(user.getUsername())
```
Comment (WesleyClements) [inline backend/authentication-microservice/src/main/java/com/ss/utopia/auth/entity/User.java:158]: Where are we using equals or hashCode?

Comment (popsps) [reply backend/authentication-microservice/src/main/java/com/ss/utopia/auth/entity/User.java:158]: assertEquals Users

File: backend/authentication-microservice/src/test/java/com/ss/utopia/auth/controller/SessionControllerIntegrationTest.java (hunk 3 above)
Comment (popsps) [reply backend/authentication-microservice/src/test/java/com/ss/utopia/auth/controller/SessionControllerIntegrationTest.java:58]: No, it is a helper function

File: backend/authentication-microservice/src/main/java/com/ss/utopia/auth/controller/SessionController.java (hunk 1 above)
Comment (popsps) [reply backend/authentication-microservice/src/main/java/com/ss/utopia/auth/controller/SessionController.java:101]: Added per Patrick's request.

**Review by Samnj583 — APPROVED** [review]

**Review by WesleyClements — APPROVED** [review]

**Pull request merged**